test_util/marathon.py

```diff
@@ -200,11 +200,46 @@ def _pool_for_marathon_app(app_id):
                 return None
 
         try:
-            return _pool_for_marathon_app(app_definition['id'])
+            return _poll_marathon_for_app_deployment(app_definition['id'])
         except retrying.RetryError:
             raise Exception("Application deployment failed - operation was not "
                             "completed in {} seconds.".format(timeout))
 
+    def ensure_deployments_complete(self, timeout=600):
+        """
+        This method ensures that, there are no pending deployments
+
+        :return: True if all deployments are completed within time out. Raises an exception otherwise.
+        """
+
+        @retrying.retry(wait_fixed=1000, stop_max_delay=120 * 1000, retry_on_exception=lambda x: False)
+        def _get_deployments_json():
+            r = self.get('v2/deployments')
+            r.raise_for_status()
+            return r.json()
+
+        def retry_on_assertion_error(exception):
+            return isinstance(exception, AssertionError)
+
+        @retrying.retry(retry_on_exception=retry_on_assertion_error,
+                        stop_max_attempt_number=10,
+                        wait_fixed=timeout * 1000)
+        def ensure_deployment_is_finished():
+            deployments_json = _get_deployments_json()
+            logging.info("v2/deployments json: {deployments_json}".format(deployments_json=deployments_json))
+            assert not deployments_json, "No deployment should be happening."
+
+        try:
+            ensure_deployment_is_finished()
+        except retrying.RetryError:
+            raise Exception("Deployments were not completed within {timeout} seconds".format(timeout=timeout))
+
+    def deploy_app_and_wait_for_deployment(self, app_definition, timeout=120):
```

Comment:
1. On `ensure_deployments_complete`. The reason for 10-minute timeout vs 2-minute timeout was that individual apps have a 2-minute default timeout, whereas this method should be called after a number of apps have been deployed together and thus will need longer time to ensure that all apps have been deployed. 

2. Since `deploy_app_and_wait_for_deployment` is for a single app,  it is using the same timeout as `deploy-app` (120 secs).  I ensured that part.

> What if we want to block on the deploy, and allow for tasks to fail during it?

I imagine that will be a deployment failure. Isn't it? 

> If we need to provide deploy-and-block behavior, I think it'd make more sense as an optional deploy_timeout

Hmm. This seems to be introducing a special case of `deployment` verification under timeout flag.

I agree with the different questions on kwargs that you raised, I faced the same problem and tried to make applicable choices with respect to deployment while reusing the existing API.

If the usage of this new API method is not intuitive enough, I think, I would go back to the previous state of this PR.

That is

* Keep ensure_deployments_complete as a separate method (which already is), but with the timeout as 120. Let the client customize it.
* Let the client call `ensure_deployments_complete` after calling the `deploy_app` method. It's easy to know the intention while reading the code too.

Is that alright with you and @BenWhitehead  ?